web/src/billing/event_status.ts

```diff
@@ -125,30 +169,33 @@ export async function stripe_payment_intent_status_check(stripe_payment_intent_i
     }
 }
 
-export async function check_status() {
+export async function check_status(): Promise<boolean> {
     if ($("#data").attr("data-stripe-session-id")) {
         return await stripe_checkout_session_status_check(
-            $("#data").attr("data-stripe-session-id"),
+            $("#data").attr("data-stripe-session-id")!,
         );
     }
     return await stripe_payment_intent_status_check(
-        $("#data").attr("data-stripe-payment-intent-id"),
+        $("#data").attr("data-stripe-payment-intent-id")!,
     );
 }
 
-async function start_status_polling() {
-    let completed = false;
+async function start_status_polling(): Promise<void> {
+    let completed: boolean = false;
     try {
         completed = await check_status();
     } catch {
-        setTimeout(start_status_polling, 5000);
+        // If the request fails, we just retry after 5 seconds.
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await start_status_polling();
```

Comment:
This change was mainly done due to eslint complaining about the rule "no-misused-promises" though it can also be avoided by just placing a void keyword before `start_status_polling` in `setTimeout` like this:
`setTimeout(() => void start_status_polling(), 5000);`
I think this might bit be more efficient than doing it like above change?

Reply:
Yeah, this looks cleaner.